src/backend/distributed/planner/multi_logical_optimizer.c

```diff
@@ -1616,7 +1616,12 @@ MasterAggregateExpression(Aggref *originalAggregate,
 		Expr *directarg;
 		foreach_ptr(directarg, originalAggregate->aggdirectargs)
 		{
-			if (!IsA(directarg, Const) && !IsA(directarg, Param))
+			/*
+			 * Need to replace nodes that contain any Vars with Vars referring
+			 * to the related column of the result set returned for the worker
+			 * aggregation.
```

Comment:
why though?

it might be good to comment on why we skip when there are no Vars, e.g.

When there are no Vars, then the expression can be fully evaluated on the coordinator, so we skip it here. This is not just an optimization, since the result of the expression might be needed to call the final function of the aggregate even when there are no input rows.